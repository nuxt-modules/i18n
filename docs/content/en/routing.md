---
title: Routing
description: "**@nuxtjs/i18n** overrides Nuxt default routes to add locale prefixes to every URL (except in no_prefix strategy)."
position: 6
category: Guide
---

**@nuxtjs/i18n** overrides Nuxt default routes to add locale prefixes to every URL (except in `no_prefix` strategy).
Say your app supports two languages: French and English as the default language, and you have the following pages in your project:

```asciidoc
pages/
├── index.vue
├── about.vue
```

This would result in the following routes being generated

```js
[
  {
    path: "/",
    component: _3237362a,
    name: "index___en"
  },
  {
    path: "/fr/",
    component: _3237362a,
    name: "index___fr"
  },
  {
    path: "/about",
    component: _71a6ebb4,
    name: "about___en"
  },
  {
    path: "/fr/about",
    component: _71a6ebb4,
    name: "about___fr"
  }
]
```

Note that routes for the English version do not have any prefix because it is the default language, see [strategies](/strategies) for more details.
