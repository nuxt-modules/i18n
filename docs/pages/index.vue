<script setup lang="ts">
import { useClipboard } from '@vueuse/core'
const { data: page } = await useAsyncData('index', () => queryContent('/').findOne())

// Page Metadata (SEO & OG)
const { setPageMeta } = usePageMeta()
setPageMeta({
  title: page.value.title,
  description: page.value.description,
  headline: page.value.hero.headline?.label
})
const source = ref('npx nuxi module add i18n')
const { copy, copied } = useClipboard({ source })
</script>

<template>
  <div>
    <ULandingHero
      v-if="page.hero"
      v-bind="page.hero"
      :ui="{
        container: 'overflow-hidden py-10 flex flex-row items-center  gap-1',
        links: 'flex items-center gap-2',
        description: 'text-gray-500 dark:text-gray-400 text-xl max-w-2xl leading-normal mb-10'
      }"
    >
      <template #headline>
        <UBadge v-if="page.hero.headline" variant="subtle" size="lg" class="relative rounded-full font-semibold">
          <NuxtLink :to="page.hero.headline.to" target="_blank" class="focus:outline-none" tabindex="-1">
            <span class="absolute inset-0" aria-hidden="true" />
          </NuxtLink>

          {{ page.hero.headline?.label }}

          <UIcon
            v-if="page.hero.headline.icon"
            :name="page.hero.headline.icon"
            class="ml-1 w-4 h-4 pointer-events-none"
          />
        </UBadge>
      </template>

      <template #description>
        {{ page.hero.description }}
      </template>

      <template #title>
        <p>
          <span class="md:hidden">i18n</span>
          <span class="hidden md:block">Internationalization</span>
          for
          <span class="text-primary">Nuxt Applications</span>
        </p>
      </template>

      <UInput
        aria-label="Copy code to get started"
        :model-value="source"
        name="get-started"
        class="mx-auto"
        disabled
        autocomplete="off"
        size="lg"
        :ui="{ base: 'w-[300px] disabled:cursor-default', icon: { trailing: { pointer: '' } } }"
      >
        <template #leading>
          <UIcon name="i-ph-terminal" />
        </template>
        <template #trailing>
          <UButton
            aria-label="Copy Code"
            :color="copied ? 'green' : 'gray'"
            variant="ghost"
            :padded="false"
            :icon="copied ? 'i-ph-check' : 'i-ph-copy'"
            @click="copy(source)"
          />
        </template>
      </UInput>
    </ULandingHero>

    <ULandingSection :title="page.features.title">
      <UPageGrid>
        <ULandingCard v-for="(item, index) of page.features.items" :key="index" v-bind="item" />
      </UPageGrid>
    </ULandingSection>
  </div>
</template>
