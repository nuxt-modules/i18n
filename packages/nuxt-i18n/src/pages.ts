import createDebug from 'debug'
import { extendPages } from '@nuxt/kit'
import { localizeRoutes } from 'vue-i18n-routing'

import type { Nuxt, NuxtPage } from '@nuxt/schema'
import type { RouteOptionsResolver, ComputedRouteOptions } from 'vue-i18n-routing'
import type { NuxtI18nOptions, CustomRoutePages } from './types'

const debug = createDebug('@nuxtjs/i18n:pages')

export function setupPages(
  options: Required<NuxtI18nOptions>,
  nuxt: Nuxt,
  additionalOptions: { isBridge?: boolean; localeCodes: string[] } = {
    isBridge: false,
    localeCodes: []
  }
) {
  let includeUprefixedFallback = nuxt.options.target === 'static'
  nuxt.hook('generate:before', () => {
    debug('called generate:before hook')
    includeUprefixedFallback = true
  })

  const pagesDir = nuxt.options.dir && nuxt.options.dir.pages ? nuxt.options.dir.pages : 'pages'
  const { trailingSlash } = nuxt.options.router
  debug(`pagesDir: ${pagesDir}, tailingSlash: ${trailingSlash}`)

  extendPages(pages => {
    debug('pages making ...', pages)
    const localizedPages = localizeRoutes(pages, {
      ...options,
      includeUprefixedFallback,
      optionsResolver: getRouteOptionsResolver(options.pages, pagesDir, options.defaultLocale)
    })
    pages.splice(0, pages.length)
    pages.unshift(...(localizedPages as NuxtPage[]))
    debug('... made pages', pages)
  })
}

function getRouteOptionsResolver(
  pages: CustomRoutePages,
  pagesDir: string,
  defaultLocale: string
): RouteOptionsResolver {
  return (route, localeCodes): ComputedRouteOptions | null => {
    const options: ComputedRouteOptions = {
      locales: localeCodes,
      paths: {}
    }
    const pattern = new RegExp(`${pagesDir}/`, 'i')
    // TODO: we might be needed support for vite, this is for webpack
    const chunkName = route.chunkName ? route.chunkName.replace(pattern, '') : route.name
    const pageOptions = chunkName ? pages[chunkName] : undefined
    // Routing disabled
    if (pageOptions === false) {
      return null
    }
    // Skip if no page options defined
    if (!pageOptions) {
      return options
    }

    // Remove disabled locales from page options
    options.locales = options.locales.filter(locale => pageOptions[locale] !== false)

    // Construct paths object
    for (const locale of options.locales) {
      const customLocalePath = pageOptions[locale]
      if (typeof customLocalePath === 'string') {
        // Set custom path if any
        options.paths[locale] = customLocalePath
        continue
      }

      const customDefaultLocalePath = pageOptions[defaultLocale]
      if (typeof customDefaultLocalePath === 'string') {
        // Set default locale's custom path if any
        options.paths[locale] = customDefaultLocalePath
      }
    }

    return options
  }
}
