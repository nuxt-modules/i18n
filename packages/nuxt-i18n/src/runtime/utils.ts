import { isVue3 } from 'vue-demi'
import { useRequestHeaders } from '#app'
import { findBrowserLocale } from 'vue-i18n-routing'

import type { Composer } from '@intlify/vue-i18n-bridge'
import type { NuxtI18nInternalOptions } from '#build/i18n.options.mjs'

/**
 * Parses locales provided from browser through `accept-language` header.
 *
 * @param input - Accept-Language header value.
 * @return An array of locale codes. Priority determined by order in array.
 */
export function parseAcceptLanguage(input: string): string[] {
  // Example input: en-US,en;q=0.9,nb;q=0.8,no;q=0.7
  // Contains tags separated by comma.
  // Each tag consists of locale code (2-3 letter language code) and optionally country code
  // after dash. Tag can also contain score after semicolon, that is assumed to match order
  // so it's not explicitly used.
  return input.split(',').map(tag => tag.split(';')[0])
}

export async function loadAndSetLocale(newLocale: string, i18n: Composer /*, { initialSetup = false } = {}*/) {
  if (!newLocale) {
    return
  }

  // abort if different domains option enabled
  // if (!initialSetup && i18n.differentDomains) {
  //   return
  // }

  const oldLocale = i18n.locale.value as string
  if (newLocale === oldLocale) {
    return
  }

  i18n.locale.value = newLocale
  console.log('loadAndSetLocale', newLocale, oldLocale, i18n)
}

export function getBrowserLocale(options: Required<NuxtI18nInternalOptions>, context?: any): string | undefined {
  let ret: string | undefined

  if (process.client) {
    if (navigator.languages) {
      // get browser language either from navigator if running on client side, or from the headers
      ret = findBrowserLocale(options.__normalizedLocales, navigator.languages as string[])
    }
  } else if (process.server) {
    if (!isVue3) {
      if (context.req && typeof context.req.headers['accept-language'] !== 'undefined') {
        ret = findBrowserLocale(
          options.__normalizedLocales,
          parseAcceptLanguage(context.req.headers['accept-language'])
        )
      }
    } else {
      const headers = useRequestHeaders(['accept-language'])
      if (headers['accept-language']) {
        ret = findBrowserLocale(options.__normalizedLocales, parseAcceptLanguage(headers['accept-language']))
      }
    }
  }

  return ret
}
