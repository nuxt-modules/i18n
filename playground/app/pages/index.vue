<template>
  <div>
    <h1>{{ $t('welcome') }}</h1>
  </div>
</template>
