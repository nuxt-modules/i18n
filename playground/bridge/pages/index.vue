<script setup lang="ts">
import { useI18n } from 'vue-i18n-bridge'
import { localePath, switchLocalePath, useI18nHead } from 'vue-i18n-routing'

const { t, locale, locales, getBrowserLocale } = useI18n()
const i18nHead = useI18nHead({ addSeoAttributes: true })

console.log('getBrowserLocale', getBrowserLocale())

useMeta({
  title: computed(() => t('hello', { name: 'nuxt bridge' })),
  htmlAttrs: computed(() => ({
    lang: i18nHead.value.htmlAttrs!.lang
  })),
  link: computed(() => [...(i18nHead.value.link || [])]),
  meta: computed(() => [...(i18nHead.value.meta || [])])
})

function getLocaleName(code) {
  const locale = locales.value.find(i => i.code === code)
  return locale ? locale.name : code
}

const availableLocales = computed(() => {
  return locales.value.filter(i => i.code !== locale.value)
})
</script>

<template>
  <div>
    <h1>Demo: Nuxt Bridge</h1>
    <h2>{{ $t('hello', { name: 'nuxt bridge' }) }}</h2>
    <h2>Pages</h2>
    <nav>
      <NuxtLink :to="localePath('/')">Home</NuxtLink> |
      <NuxtLink :to="localePath('/about')">About</NuxtLink>
    </nav>
    <h2>Current Language: {{ getLocaleName(locale) }}</h2>
    <h2>Select Languages</h2>
    <nav>
      <span v-for="locale in availableLocales" :key="locale.code">
        <NuxtLink :to="switchLocalePath(locale.code) || ''">{{ locale.name }}</NuxtLink> |
      </span>
    </nav>
    <h2>I18n Head</h2>
    <div>
      <p>{{ i18nHead }}</p>
    </div>
  </div>
</template>
