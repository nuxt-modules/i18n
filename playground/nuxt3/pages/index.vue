<script setup lang="ts">
import { useI18n } from 'vue-i18n'
import { localePath, switchLocalePath, useI18nHead, useBrowserLocale } from '#i18n'

const { t, locale, locales } = useI18n()

console.log('useBrowserLocale', useBrowserLocale())

const i18nHead = useI18nHead({ addSeoAttributes: true })
useMeta({
  title: computed(() => t('hello', { name: 'nuxt bridge' })),
  htmlAttrs: computed(() => ({
    lang: i18nHead.value.htmlAttrs!.lang
  })),
  link: computed(() => [...(i18nHead.value.link || [])]),
  meta: computed(() => [...(i18nHead.value.meta || [])])
})

function getLocaleName(code) {
  const locale = locales.value.find(i => i.code === code)
  return locale ? locale.name : code
}

const availableLocales = computed(() => {
  return locales.value.filter(i => i.code !== locale.value)
})
</script>

<template>
  <div>
    <h1>Demo: Nuxt 3</h1>
    <h2>{{ $t('hello', { name: 'nuxt3' }) }}</h2>
    <h2>Pages</h2>
    <nav>
      <NuxtLink :to="localePath('/')">Home</NuxtLink> |
      <NuxtLink :to="localePath('/about')">About</NuxtLink>
    </nav>
    <h2>Current Language: {{ getLocaleName(locale) }}</h2>
    <h2>Select Languages</h2>
    <nav>
      <span v-for="locale in availableLocales" :key="locale.code">
        <NuxtLink :to="switchLocalePath(locale.code) || ''">{{ locale.name }}</NuxtLink> |
      </span>
    </nav>
    <h2>I18n Head</h2>
    <div>
      <p>{{ i18nHead }}</p>
    </div>
  </div>
</template>
