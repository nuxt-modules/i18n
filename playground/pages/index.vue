<script setup lang="ts">
import { computed } from 'vue'

// import { useLocalePath, useSwitchLocalePath, useLocaleHead, useBrowserLocale } from '#i18n'
import { LocaleObject } from '#i18n'

const route = useRoute()
const { t, strategy, locale, locales, localeProperties, setLocale, finalizePendingLocaleChange } = useI18n()
const localePath = useLocalePath()
const switchLocalePath = useSwitchLocalePath()
const getRouteBaseName = useRouteBaseName()

// route.meta.pageTransition.onBeforeEnter = async () => {
//   await finalizePendingLocaleChange()
// }

console.log('route base name', getRouteBaseName())
console.log('useBrowserLocale', useBrowserLocale())
console.log('localeProperties', localeProperties)
console.log('foo', t('foo'))
console.log('message if local layer merged:', t('layerText'))
console.log('message if github layer merged:', t('layer-test-key'))

function getLocaleName(code: string) {
  const locale = (locales.value as LocaleObject[]).find(i => i.code === code)
  return locale ? locale.name : code
}

const availableLocales = computed(() => {
  return (locales.value as LocaleObject[]).filter(i => i.code !== locale.value)
})

definePageMeta({
  title: 'pages.title.top',
  middleware: () => {
    const localePath2 = useLocalePath()
    console.log('middleware', localePath2({ name: 'blog' }))
  },
  pageTransition: {
    name: 'page',
    mode: 'out-in',
    onBeforeEnter: async () => {
      const { finalizePendingLocaleChange } = useNuxtApp().$i18n
      await finalizePendingLocaleChange()
      console.log('onBeforeEnter')
    }
  }
})
</script>

<template>
  <div>
    <h1>Demo: Nuxt 3</h1>
    <h2>{{ $t('hello', { name: 'nuxt3' }) }}</h2>
    <p>{{ $t('bar.buz', { name: 'buz' }) }}</p>
    <h2>Pages</h2>
    <nav>
      <NuxtLink :to="localePath('/')">Home</NuxtLink> | <NuxtLink :to="localePath({ name: 'about' })">About</NuxtLink> |
      <NuxtLink :to="localePath({ name: 'blog' })">Blog</NuxtLink> |
      <NuxtLink :to="localePath({ name: 'category-id', params: { id: 'foo' } })">Category</NuxtLink> |
      <NuxtLink :to="localePath({ name: 'history' })">History</NuxtLink>
    </nav>
    <h2>Current Language: {{ getLocaleName(locale) }}</h2>
    <h2>Current Strategy: {{ strategy }}</h2>
    <h2>Select Languages with switchLocalePath</h2>
    <nav>
      <span v-for="locale in availableLocales" :key="locale.code">
        <NuxtLink :to="switchLocalePath(locale.code) || ''">{{ locale.name }}</NuxtLink> |
      </span>
    </nav>
    <h2>Select Languages with setLocale</h2>
    <nav>
      <span v-for="locale in availableLocales" :key="locale.code">
        <a href="javascript:void(0)" @click="setLocale(locale.code)">{{ locale.name }}</a> |
      </span>
    </nav>
    <p>{{ $t('settings.profile') }}</p>
    <p>{{ $t('tag') }}</p>
  </div>
</template>

<style scoped>
.page-enter-active,
.page-leave-active {
  transition: opacity 1s;
}
.page-enter,
.page-leave-active {
  opacity: 0;
}
</style>
