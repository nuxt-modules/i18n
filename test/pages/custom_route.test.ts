import { vi, test, expect } from 'vitest'
import fs from 'node:fs'
import { resolve } from 'node:path'
import { localizeRoutes } from '../../src/routing'
import { getRouteOptionsResolver, analyzeNuxtPages } from '../../src/pages'
import { getNuxtOptions, stripFilePropertyFromPages } from './utils'

import type { NuxtPage } from '@nuxt/schema'
import type { NuxtPageAnalyzeContext, AnalyzedNuxtPageMeta } from '../../src/pages'
import type { NuxtI18nOptions } from '../../src/types'

/**
 * NOTE:
 *  This teardown is to avoid the weird issue of vitest.
 *  https://github.com/vitest-dev/vitest/issues/2845#issuecomment-1424666525
 */
afterAll(() => {
  vi.restoreAllMocks()
})

describe.each([
  {
    case: 'simple',
    options: getNuxtOptions({
      about: {
        en: '/about-us',
        fr: '/a-propos',
        es: '/sobre'
      }
    }),
    pages: [
      {
        path: '/about',
        file: '/path/to/nuxt-app/pages/about.vue',
        children: []
      }
    ]
  },
  {
    case: 'the part of URL',
    options: getNuxtOptions({
      about: {
        fr: '/a-propos'
      },
      'services/index': {
        fr: '/offres'
      },
      'services/development/index': {
        fr: '/offres/developement'
      },
      'services/development/app': {
        fr: '/offres/developement/app'
      },
      'services/development/website': {
        fr: '/offres/developement/site-web'
      },
      'services/coaching': {
        fr: '/offres/formation'
      }
    }),
    pages: [
      {
        path: '/about',
        file: '/path/to/nuxt-app/pages/about.vue',
        children: []
      },
      {
        name: 'services-coaching',
        path: '/services/coaching',
        file: '/path/to/nuxt-app/pages/services/coaching.vue',
        children: []
      },
      {
        name: 'services-development-app',
        path: '/services/development/app',
        file: '/path/to/nuxt-app/pages/services/development/app.vue',
        children: []
      },
      {
        name: 'services-development',
        path: '/services/development',
        file: '/path/to/nuxt-app/pages/services/development/index.vue',
        children: []
      },
      {
        name: 'services-development-website',
        path: '/services/development/website',
        file: '/path/to/nuxt-app/pages/services/development/website.vue',
        children: []
      },
      {
        name: 'services',
        path: '/services',
        file: '/path/to/nuxt-app/pages/services/index.vue',
        children: []
      }
    ]
  },
  {
    case: 'dynamic parameters',
    options: getNuxtOptions({
      'blog/[date]/[slug]': {
        ja: '/blog/tech/[date]/[slug]'
      }
    }),
    pages: [
      {
        name: 'blog-date-slug',
        path: '/blog/:date/:slug',
        file: '/path/to/nuxt-app/pages/blog/[date]/[slug].vue',
        children: []
      }
    ]
  }
])('Module configuration', ({ case: _case, options, pages }) => {
  test(_case, () => {
    vi.spyOn(fs, 'readFileSync').mockReturnValue('')

    const srcDir = '/path/to/nuxt-app'
    const pagesDir = 'pages'
    const ctx: NuxtPageAnalyzeContext = {
      stack: [],
      srcDir,
      pagesDir,
      pages: new Map<NuxtPage, AnalyzedNuxtPageMeta>()
    }

    analyzeNuxtPages(ctx, pages)
    const localizedPages = localizeRoutes(pages, {
      ...options,
      includeUnprefixedFallback: false,
      optionsResolver: getRouteOptionsResolver(ctx, options as Required<NuxtI18nOptions>)
    } as Parameters<typeof localizeRoutes>[1])

    expect(localizedPages).toMatchSnapshot()
  })
})

describe.each([
  {
    case: 'simple',
    options: getNuxtOptions({}, 'page'),
    pages: [
      {
        path: '/about',
        file: resolve(__dirname, '../fixtures/custom_route/simple/pages/about.vue'),
        children: []
      }
    ]
  },
  {
    case: 'dynamic route',
    options: getNuxtOptions({}, 'page'),
    pages: [
      {
        name: 'articles-name',
        path: '/articles/:name',
        file: resolve(__dirname, '../fixtures/custom_route/dynamic/pages/articles/[name].vue'),
        children: []
      }
    ]
  },
  {
    case: 'with definePageMeta',
    options: getNuxtOptions({}, 'page'),
    pages: [
      {
        path: '/about',
        file: resolve(__dirname, '../fixtures/custom_route/with_meta/pages/about.vue'),
        children: []
      }
    ]
  },
  {
    case: 'JavaScript',
    options: getNuxtOptions({}, 'page'),
    pages: [
      {
        path: '/about',
        file: resolve(__dirname, '../fixtures/custom_route/js/pages/about.vue'),
        children: []
      }
    ]
  }
])('Page components', ({ case: _case, options, pages }) => {
  test(_case, () => {
    const srcDir = '/path/to/nuxt-app'
    const pagesDir = 'pages'
    const ctx: NuxtPageAnalyzeContext = {
      stack: [],
      srcDir,
      pagesDir,
      pages: new Map<NuxtPage, AnalyzedNuxtPageMeta>()
    }

    analyzeNuxtPages(ctx, pages)
    const localizedPages = localizeRoutes(pages, {
      ...options,
      includeUnprefixedFallback: false,
      optionsResolver: getRouteOptionsResolver(ctx, options as Required<NuxtI18nOptions>)
    } as Parameters<typeof localizeRoutes>[1])
    expect(stripFilePropertyFromPages(localizedPages)).toMatchSnapshot()
  })
})

test('#1649', () => {
  const pages = [
    {
      path: '/account',
      file: '/path/to/1649/pages/account.vue',
      children: [
        {
          name: 'account-addresses',
          path: 'addresses',
          file: '/path/to/1649/pages/account/addresses.vue',
          children: []
        },
        {
          name: 'account',
          path: '',
          file: '/path/to/1649/pages/account/index.vue',
          children: []
        },
        {
          name: 'account-profile',
          path: 'profile',
          file: '/path/to/1649/pages/account/profile.vue',
          children: []
        }
      ]
    },
    {
      name: 'index',
      path: '/',
      file: '/path/to/1649/pages/index.vue',
      children: []
    }
  ]

  const options = getNuxtOptions({
    account: {
      fr: '/compte'
    },
    'account/profile': {
      fr: '/compte/profil'
    },
    'account/addresses': {
      fr: '/compte/adresses'
    }
  })

  vi.spyOn(fs, 'readFileSync').mockReturnValue('')

  const srcDir = '/path/to/1649'
  const pagesDir = 'pages'
  const ctx: NuxtPageAnalyzeContext = {
    stack: [],
    srcDir,
    pagesDir,
    pages: new Map<NuxtPage, AnalyzedNuxtPageMeta>()
  }

  analyzeNuxtPages(ctx, pages)
  const localizedPages = localizeRoutes(pages, {
    ...options,
    includeUnprefixedFallback: false,
    optionsResolver: getRouteOptionsResolver(ctx, options as Required<NuxtI18nOptions>)
  } as Parameters<typeof localizeRoutes>[1])

  expect(localizedPages).toMatchSnapshot()
})
