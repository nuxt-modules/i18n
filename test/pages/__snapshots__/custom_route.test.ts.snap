// Vitest Snapshot v1

exports[`#1649 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/1649/pages/account/addresses.vue",
        "name": "account-addresses___en",
        "path": "addresses",
      },
      {
        "children": [],
        "file": "/path/to/1649/pages/account/index.vue",
        "name": "account___en",
        "path": "",
      },
      {
        "children": [],
        "file": "/path/to/1649/pages/account/profile.vue",
        "name": "account-profile___en",
        "path": "profile",
      },
    ],
    "file": "/path/to/1649/pages/account.vue",
    "path": "/account",
  },
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/1649/pages/account/addresses.vue",
        "name": "account-addresses___ja",
        "path": "addresses",
      },
      {
        "children": [],
        "file": "/path/to/1649/pages/account/index.vue",
        "name": "account___ja",
        "path": "",
      },
      {
        "children": [],
        "file": "/path/to/1649/pages/account/profile.vue",
        "name": "account-profile___ja",
        "path": "profile",
      },
    ],
    "file": "/path/to/1649/pages/account.vue",
    "path": "/ja/account",
  },
  {
    "children": [
      {
        "children": [],
        "file": "/path/to/1649/pages/account/addresses.vue",
        "name": "account-addresses___fr",
        "path": "/fr/compte/adresses",
      },
      {
        "children": [],
        "file": "/path/to/1649/pages/account/index.vue",
        "name": "account___fr",
        "path": "",
      },
      {
        "children": [],
        "file": "/path/to/1649/pages/account/profile.vue",
        "name": "account-profile___fr",
        "path": "/fr/compte/profil",
      },
    ],
    "file": "/path/to/1649/pages/account.vue",
    "path": "/fr/compte",
  },
  {
    "children": [],
    "file": "/path/to/1649/pages/index.vue",
    "name": "index___en",
    "path": "/",
  },
  {
    "children": [],
    "file": "/path/to/1649/pages/index.vue",
    "name": "index___ja",
    "path": "/ja",
  },
  {
    "children": [],
    "file": "/path/to/1649/pages/index.vue",
    "name": "index___fr",
    "path": "/fr",
  },
]
`;

exports[`Modeule configration > dynamic parameters 1`] = `
[
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/blog/[date]/[slug].vue",
    "name": "blog-date-slug___en",
    "path": "/blog/:date/:slug",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/blog/[date]/[slug].vue",
    "name": "blog-date-slug___ja",
    "path": "/ja/blog/tech/:date/:slug",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/blog/[date]/[slug].vue",
    "name": "blog-date-slug___fr",
    "path": "/fr/blog/:date/:slug",
  },
]
`;

exports[`Modeule configration > simple 1`] = `
[
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/about-us",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/ja/about-us",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/fr/a-propos",
  },
]
`;

exports[`Modeule configration > the part of URL 1`] = `
[
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/about",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/ja/about",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/about.vue",
    "path": "/fr/a-propos",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/coaching.vue",
    "name": "services-coaching___en",
    "path": "/services/coaching",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/coaching.vue",
    "name": "services-coaching___ja",
    "path": "/ja/services/coaching",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/coaching.vue",
    "name": "services-coaching___fr",
    "path": "/fr/offres/formation",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/app.vue",
    "name": "services-development-app___en",
    "path": "/services/development/app",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/app.vue",
    "name": "services-development-app___ja",
    "path": "/ja/services/development/app",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/app.vue",
    "name": "services-development-app___fr",
    "path": "/fr/offres/developement/app",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/index.vue",
    "name": "services-development___en",
    "path": "/services/development",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/index.vue",
    "name": "services-development___ja",
    "path": "/ja/services/development",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/index.vue",
    "name": "services-development___fr",
    "path": "/fr/offres/developement",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/website.vue",
    "name": "services-development-website___en",
    "path": "/services/development/website",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/website.vue",
    "name": "services-development-website___ja",
    "path": "/ja/services/development/website",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/development/website.vue",
    "name": "services-development-website___fr",
    "path": "/fr/offres/developement/site-web",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/index.vue",
    "name": "services___en",
    "path": "/services",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/index.vue",
    "name": "services___ja",
    "path": "/ja/services",
  },
  {
    "children": [],
    "file": "/path/to/nuxt-app/pages/services/index.vue",
    "name": "services___fr",
    "path": "/fr/offres",
  },
]
`;

exports[`Page components > JavaScript 1`] = `
[
  {
    "children": [],
    "path": "/about-us",
  },
  {
    "children": [],
    "path": "/ja/about-ja",
  },
  {
    "children": [],
    "path": "/fr/a-propos",
  },
]
`;

exports[`Page components > dynamic route 1`] = `
[
  {
    "children": [],
    "name": "articles-name___en",
    "path": "/articles/:name",
  },
  {
    "children": [],
    "name": "articles-name___ja",
    "path": "/ja/%E8%A8%98%E4%BA%8B/:name",
  },
  {
    "children": [],
    "name": "articles-name___fr",
    "path": "/fr/articles/:name",
  },
]
`;

exports[`Page components > simple 1`] = `
[
  {
    "children": [],
    "path": "/about-us",
  },
  {
    "children": [],
    "path": "/ja/about-ja",
  },
  {
    "children": [],
    "path": "/fr/a-propos",
  },
]
`;

exports[`Page components > with definePageMeta 1`] = `
[
  {
    "children": [],
    "path": "/about-us",
  },
  {
    "children": [],
    "path": "/ja/about-ja",
  },
  {
    "children": [],
    "path": "/fr/a-propos",
  },
]
`;
